Panic on shutdown

When trying to exit my program, I get the following panic:

```
panic: bad file descriptor
goroutine 127 [running]:
github.com/weaveworks/scope/vendor/github.com/typetypetype/conntrack.FollowSize.func2(0x4a, 0xc4211f0300)
/go/src/github.com/weaveworks/scope/vendor/github.com/typetypetype/conntrack/client.go:168 +0xba
created by github.com/weaveworks/scope/vendor/github.com/typetypetype/conntrack.FollowSize
/go/src/github.com/weaveworks/scope/vendor/github.com/typetypetype/conntrack/client.go:159 +0x106
```

I think this is because the `stop()` function returned from `FollowSize()` just closes the netlink socket, and  doesn't take any steps to avoid `readMsgs()` returning an error at that point.

Maybe `FollowSize()` `stop()` could set a flag `closing`, and ignore errors if the flag is true?

--- a/client.go
+++ b/client.go
@@ -147,8 +147,10 @@ func Follow(flags uint32) (<-chan Conn, func(), error) {
 
 // Follow gives a channel with all changes, , using specified netlink buffer size.
 func FollowSize(bufferSize int, flags uint32) (<-chan Conn, func(), error) {
+	var closing bool
 	s, _, err := connectNetfilter(bufferSize, flags)
 	stop := func() {
+		closing = true
 		syscall.Close(s)
 	}
 	if err != nil {
@@ -164,7 +166,7 @@ func FollowSize(bufferSize int, flags uint32) (<-chan Conn, func(), error) {
 			// continue
 			// }
 			res <- c
-		}); err != nil {
+		}); err != nil && !closing {
 			panic(err)
 		}
 	}()

